src/include/os.h

```diff
@@ -139,6 +139,14 @@ struct __wt_file_handle_posix {
     int fd; /* POSIX file handle */
 
     bool direct_io; /* O_DIRECT configured */
+
+    /* The memory buffer and variables if we use mmap for I/O */
+    char *mmap_buf;
+    bool mmap_file_mappable;
+    int mmap_prot;
+    volatile uint32_t mmap_resizing;
+    size_t mmap_size;
```

Comment:
I'm suspicious `mmap_size` should be an `off_t` (or a `wt_off_t` in this case).

`mmap_buf` is probably a `uint8_t` (unless it really does point to signed characters). :)